File: Sort/selection.c
//选择排序

#include <stdio.h>

int main() {
    int a[101], n, i, j, key, temp;
    printf("请输入排序个数\n");
    scanf("%d", &n);
    for (int i = 0; i < n; i++) {
        scanf("%d", &a[i]);
    }
    for (i = 0; i < n; i++) {
        key = i;
        for (j = i + 1;  j < n; j++) {
            if (a[key] > a[j])
                key = j;
        }
        if (key != i)
        {
            temp = a[i];
            a[i] = a[key];
            a[key] = temp;
        }
    }

    for (i = 0; i < n; i++) {
        printf("%d ",a[i]);
    }
    return 0;
}

File: Sort/selection.php
<?php
/**
 * 选择排序
 * @param array $data
 * @throws Exception
 */

function selection(array $data = array()){
    if (empty($data))
        throw new Exception("请输入测试数据");
    $num = count($data);
    for ($i = 0; $i < $num; $i ++)
    {
        $key = $i;
        for ($j = $i + 1; $j < $num; $j++)
        {
            if ($data[$key] > $data[$j])
            {

                $key = $j;
            }
        }
        if ($key != $i)
        {
            $temp = $data[$key];
            $data[$key] = $data[$i];
            $data[$i] = $temp;
        }
    }
    return $data;
}


include "../Config.php";
try{
    var_dump(selection($sort_data));
}catch (Exception $e){
    var_dump($e->getMessage());
}

